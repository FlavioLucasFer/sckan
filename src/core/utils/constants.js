const MAIN_COLOR = '#1ac3d6';
const SECUNDARY_COLOR = '#245257';

export {
    MAIN_COLOR,
    SECUNDARY_COLOR,
};