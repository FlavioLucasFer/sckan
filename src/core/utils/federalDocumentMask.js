'use strict';

function federalDocumentMask(value) {
	value = value.replace(/\D/g, '');

	if (value.length <= 11) { //CPF
		value = value.replace(/(\d{3})(\d)/, '$1.$2')
			.replace(/(\d{3})(\d)/, '$1.$2')
			.replace(/(\d{3})(\d{1,2})$/, '$1-$2');
	} else { //CNPJ
		value = value.replace(/^(\d{2})(\d)/, '$1.$2')
			.replace(/^(\d{2})\.(\d{3})(\d)/, '$1.$2.$3')
			.replace(/\.(\d{3})(\d)/, '.$1/$2')
			.replace(/(\d{4})(\d)/, '$1-$2')
			.replace(/(-\d{2})\d+?$/, '$1');
	}

	return value;
}

export default federalDocumentMask;
