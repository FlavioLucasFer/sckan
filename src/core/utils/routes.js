const HOME_ROUTE = '/';
const LOGIN_ROUTE = '/login';
const COMPANY_ROUTE = '/companies';
const COMPANY_FORM_ROUTE = `${COMPANY_ROUTE}/form`;
const PROJECT_ROUTE = '/projects';
const PROJECT_FORM_ROUTE = `${PROJECT_ROUTE}/form`;
const SPRINT_ROUTE = '/sprints';
const SPRINT_FORM_ROUTE = `${SPRINT_ROUTE}/form`;
const TASK_ROUTE = '/tasks';
const TASK_FORM_ROUTE = `${TASK_ROUTE}/form`;
const USER_ROUTE = '/users';
const USER_FORM_ROUTE = `${USER_ROUTE}/form`;
const USER_PROFILE = '/user';
const USER_SITTINGS = `${USER_PROFILE}/settings`;

export {
	HOME_ROUTE,
  LOGIN_ROUTE,
  COMPANY_ROUTE,
	COMPANY_FORM_ROUTE,
  PROJECT_ROUTE,
	PROJECT_FORM_ROUTE,
  SPRINT_ROUTE,
	SPRINT_FORM_ROUTE,
  TASK_ROUTE,
	TASK_FORM_ROUTE,
  USER_ROUTE,
	USER_FORM_ROUTE,
	USER_PROFILE,
	USER_SITTINGS,
};