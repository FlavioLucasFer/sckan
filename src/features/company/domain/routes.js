'use strict';

const { REACT_APP_API_URL } = process.env;

const BASE_URL = `${REACT_APP_API_URL}/companies`;

const ROUTES = {
	CREATE: `${BASE_URL}/create`,
	EDIT: `${BASE_URL}/edit`,
	FIND: `${BASE_URL}/find`,
	FIND_BY_ID: `${BASE_URL}/find/:id`,
	REMOVE: `${BASE_URL}/remove`,
};

export default ROUTES;
