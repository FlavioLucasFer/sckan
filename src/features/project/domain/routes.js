'use strict'

const { REACT_APP_API_URL } = process.env;

const BASE_URL = `${REACT_APP_API_URL}/projects`;

const ROUTES = {
	CREATE: `${BASE_URL}/create`,
	EDIT: `${BASE_URL}/edit`,
	REMOVE: `${BASE_URL}/remove`,
	FIND: `${BASE_URL}/find`,
};

export default ROUTES;
