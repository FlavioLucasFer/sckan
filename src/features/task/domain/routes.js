'use strict'

const { REACT_APP_API_URL } = process.env;

const BASE_URL = `${REACT_APP_API_URL}/tasks`;

const ROUTES = {
	CREATE: `${BASE_URL}/create`,
	EDIT: `${BASE_URL}/edit`,
	REMOVE: `${BASE_URL}/remove`,
	FIND: `${BASE_URL}/find`,
	FIND_BY_ID: `${BASE_URL}/find/:id`,
};

export default ROUTES;
