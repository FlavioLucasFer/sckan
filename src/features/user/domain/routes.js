'use strict'

const { REACT_APP_API_URL } = process.env;

const BASE_URL = `${REACT_APP_API_URL}/users`;

const ROUTES = {
	CREATE: `${BASE_URL}/create`,
	EDIT: `${BASE_URL}/edit`,
	REMOVE: `${BASE_URL}/remove`,
	FIND: `${BASE_URL}/find`,
	FIND_BY_ID: `${BASE_URL}/find/:id`,
	SIGN_IN: `${BASE_URL}/sign-in`,
	LOGOUT: `${BASE_URL}/logout`,
};

export default ROUTES;
