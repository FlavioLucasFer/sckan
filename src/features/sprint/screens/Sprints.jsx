import React, { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import ListScreen from 'core/components/ListScreen';
import useFormInput from 'core/hooks/useFormInput';
import Select from 'core/components/Select';
import { SPRINT_FORM_ROUTE } from 'core/utils/routes';

import projectService from 'features/project/domain/service';
import sprintService from 'features/sprint/domain/service';

import { setProjects } from 'redux/reducers/projects';

function Sprints() {
	const dispatch = useDispatch();
	const [contentMessage, setContentMessage] = useState('Select some project and click on filter button!');
	const [sprints, setSprints] = useState([]);
	const company = useSelector(state => state.company.value);
	const projects = useSelector(state => state.projects.value);
	const project = useFormInput();

	useEffect(async () => {
		try {
			const projects = await projectService.find({ company: company.id });
			dispatch(setProjects(projects));
		} catch (err) {
			console.log('err:', err);
		}
	}, []);
	
	const fields = [
		{ title: 'Description', field: 'description' },
		{ title: 'Duration', field: 'sprintDuration' },
	];

	function renderFiltersRow() {
		return [
			<Select {...project} 
				key="s1"
				title="Project"
				values={projects}
				className="col s12 m4 l3"
				valueAs="id"
				titleAs="name"
				defaultOption="Select a project" />
		];
	}

	async function getSprints() {
		try {
			const sprints = await sprintService.find({ project: project.value });
			setSprints(sprints);
		} catch (err) {
			throw err;
		}
	}

	async function handleFilter() {
		try {
			await getSprints();
			setContentMessage('');
		} catch (err) {
			console.log('err:', err);
		}
	}

	async function handleRemove(id) {
		try {
			await sprintService.remove(id);
			await getSprints();
		} catch (err) {
			console.log('err:', err);
		}
	}

  return (
		<ListScreen title="Sprints"
			singularTitle="Sprint"
			data={sprints}
			fields={fields}
			identifierField="customIdentifier"
			itemsQuantityByPage={10}
			formRoute={SPRINT_FORM_ROUTE}
			filters={renderFiltersRow()}
			filterButtonDisabled={!project.value}
			contentMessage={contentMessage}
			onRefresh={getSprints}
			onRemove={handleRemove}
			onFilter={handleFilter} />
  );
}

export default Sprints;
