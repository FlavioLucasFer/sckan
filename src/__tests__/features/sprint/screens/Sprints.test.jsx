import React from 'react';
import { shallow } from 'enzyme';

import Sprints from 'features/sprint/screens/Sprints';
import { SPRINT_FORM_ROUTE } from 'core/utils/routes';

const mockHistory = {
	push: jest.fn(),
};

jest.mock('react-router-dom', () => ({
	...(jest.requireActual('react-router-dom')),
	useHistory: () => (mockHistory),
}));

describe('Testing Sprints screen component', () => {
	afterAll(() => {
		jest.clearAllMocks();
	});

	it('should render correctly', () => {
		const wrapper = shallow(
			<Sprints />
		);

		expect(wrapper.debug()).toMatchSnapshot();
	});

	it('should navigate to sprint form screen when clink on "add" button', () => {
		const wrapper = shallow(
			<Sprints />
		);

		wrapper.find({ formRoute: SPRINT_FORM_ROUTE })
			.dive()
			.find('.new-item-button')
			.simulate('click')

		expect(mockHistory.push).toBeCalledWith(SPRINT_FORM_ROUTE);
	});
});
