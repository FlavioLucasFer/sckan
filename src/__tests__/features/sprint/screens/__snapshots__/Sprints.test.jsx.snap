// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Sprints screen component should render correctly 1`] = `"<ListScreen title=\\"Sprints\\" singularTitle=\\"Sprint\\" data={{...}} fields={{...}} identifierField=\\"\\" pagesQuantity={1} formRoute=\\"/sprints/form\\" onRefresh={{...}} />"`;
